### PR TITLE
Clone core

### PR DESCRIPTION
**Pull request by xoddong**
-Edited @spasiu 's work to communicate with front end
-Added associations between team and library

### CODE REVIEW

File: api/controllers/TeamController.js [hunk 1]
```diff
@@ -5,7 +5,16 @@
...
+
 module.exports = {
-	
-};
+  create: function(req, res) {
+    var team = req.params.all();
 
+    Team.create(team).then(function(newTeam){
+      Core.clone(newTeam, function(lib){
+        res.send({"teamId": newTeam.id})
```
Comment (jplouie) [inline api/controllers/TeamController.js]: ;

File: api/controllers/TeamController.js [hunk 2]
```diff
@@ -5,7 +5,16 @@
...
 
+var Core = require("../services/CoreLibrary.js");
+
 module.exports = {
-	
-};
+  create: function(req, res) {
+    var team = req.params.all();
 
+    Team.create(team).then(function(newTeam){
```
Comment (jplouie) [inline api/controllers/TeamController.js]: needs a space before the {

File: api/controllers/TeamController.js [hunk 3]
```diff
@@ -5,7 +5,16 @@
...
+      Core.clone(newTeam, function(lib){
```
Comment (jplouie) [inline api/controllers/TeamController.js]: space here too

File: api/services/coreLibrary.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        module: moduleId
+      });
+    };
+
+    var cloneModule = function(libraryId, coreModule, callback) {
+      Module.create({
+        name: coreModule.name,
+        library: libraryId
+      })
+      .then(function(newModule){
```
Comment (jplouie) [inline api/services/coreLibrary.js]: space here

File: api/services/coreLibrary.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        nModule = newModule;
+        return coreModule.lessons.map(function(coreLesson){
```
Comment (jplouie) [inline api/services/coreLibrary.js]: space

**Comment by xoddong** [comment]
so because right now when you create a team, it doesn't automatically route anywhere, to test this out:
1. go to coreLibrary service
2. around line 58 in the function that says -> 
 .spread(function(clib, nlib) {
      console.log(nlib); <===== console.log this
      // Arugments are: core library, new library, and all the modules associated with core library. 
      // Array.prototype.slice.call will only return the values of the arguments;
      var modules = Array.prototype.slice.call(arguments);
      var modules = modules.slice(2, modules.length);


so when you create the team, your console should have the newly created library (and thus, it's id). Manually route to that library id to test if the library has been created.

**Comment by Conroyce** [comment]
code works!

**Comment by kennyczadzeck** [comment]
Code works for me as well.

**Comment by delasky** [comment]
waiting for fix to lesson clone

**Pull request merged**